src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -140,13 +139,13 @@
         List<Path> reportFoldersAndFiles = cloneAndAnalyzeRepos(configs, outputPath,
                 numCloningThreads, numAnalysisThreads, shouldFreshClone);
 
-        Date reportSinceDate = (cliSinceDate.equals(SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE))
+        LocalDateTime reportSinceDate = (cliSinceDate.equals(SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE))
                 ? earliestSinceDate : cliSinceDate;
 
         Optional<Path> summaryPath = FileUtil.writeJsonFile(
                 new SummaryJson(configs, reportConfig, generationDate,
-                        TimeUtil.getZonedDateFromSystemDate(reportSinceDate, zoneId),
-                        TimeUtil.getZonedDateFromSystemDate(untilDate, zoneId), isSinceDateProvided,
+                        reportSinceDate,
+                        untilDate, isSinceDateProvided,
```

Comment:
I think this should be using the time zone given in zoneId? We are generating the report based on the user's given time zone.

Reply:
I'm not sure what you mean here since `reportSinceDate` and `untilDate` are `LocalDateTime` objects. The timezone is not fixed in the code but there is an implicit assumption that `reportSinceDate` and `untilDate` are with reference to the timezone of `zoneId`, which is also passed into the constructor.

Reply:
The `reportSinceDate`, if it uses `cliSinceDate`, then it is correct as it is implicitly assumed as the user's input, which will be in the time zone provided. However, if the user provides `d1` as the input, what is the time zone used?

Reply:
I just realised that the timezone in each `CommitInfo` may be different from what is given in `config`. Not only could the earliest commit date be in the wrong timezone if `d1` were used, the `LocalDateTime` stored in each `CommitResult` would also be off.

Thanks for pointing it out!

Reply:
I'm changing some lines in `CommitInfoAnalyzer` and `CommitInfoAnalyzerTest` to account for the difference in timezone.

Reply:
Can I confirm if the date obtained from `d1` is generated based on the user's given time zone given in `config.zoneId` (basically does it now correctly use the user's time zone)?

Reply:
I also noticed that for `commitInfos.isEmpty()`, `d1` would result in `getStartDate` of `CommitResultAggregator` returning a date in UTC time rather than the config timezone. I've changed it to return the `ARBITRARY_FIRST_COMMIT_DATE` in the config timezone instead.

Other than that, when `d1` is used, the date of the earliest commit date would be used for `reportSinceDate` when creating the `SummaryJson` object in `ReportGenerator`. The commit date is translated into the config timezone in `CommitInfoAnalyzer`

Reply:
Okay nice. These two lines can be merged into a single line.